model/labels/labels_common.go

```diff
@@ -38,15 +38,16 @@ type Label struct {
 	Name, Value string
 }
 
-func (ls Labels) String() string {
+func (ls Labels) toString(addSpace bool) string {
 	var b bytes.Buffer
-
 	b.WriteByte('{')
 	i := 0
 	ls.Range(func(l Label) {
 		if i > 0 {
 			b.WriteByte(',')
-			b.WriteByte(' ')
+			if addSpace {
+				b.WriteByte(' ')
```

Comment:
Removing this space only breaks two "formatting" tests. Maybe we could get rid on it in 3.0
Keeping it as it "breaks" `promtool tsdb dump`

Reply:
I would suggest moving this into promtool where you need it.
The interface of Labels is too big already.

Reply:
Could you elaborate? Do you mean redefining the same `String` on promtool side without the `b.WriteByte(' ')` line?

Reply:
`CondensedString`, yes.

Reply:
@bboreham Could you take a look now? thanks.